### PR TITLE
feat: Add indexer.idle() and initial state 'indexing'

### PR DESCRIPTION
**Pull request by gmaclennan**
This changes the initial state to `indexing`, fixing #18. It also adds an `await indexer.idle()` method, that resolves when indexing is done. Also adds more tests for various edge-cases.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -102,8 +108,20 @@ class MultiCoreIndexer extends TypedEmitter {
     this.#indexStream.addStream(coreIndexStream)
   }
 
+  /**
+   * Resolves when indexing state is 'idle'
+   */
+  async idle() {
```
Comment (achou11) [inline index.js:114]: nit but not sure about the naming of this, but can't think of a suggestion either

File: test/unit-tests/core-index-stream.test.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ const {
   create,
   replicate,
   generateFixture,
-  throttledIdle,
+  throttledDrain: throttledIdle,
```
Comment (achou11) [inline test/unit-tests/core-index-stream.test.js:10]: did you mean to rename references to throttledIdle to throttledDrain for this file? (i've run into this issue occasionally when using vscode's reference renaming feature 😄 )

**Review by achou11 — APPROVED** [review]
lgtm!

File: index.js (hunk 1 above)
Comment (gmaclennan) [reply index.js:106]: Yeah I often want to call functions like this "awaitIdle", but that just becomes confusing when you write `await awaitIdle()`
